lib/serialization/compositeSpec.ts

```diff
@@ -8,29 +8,94 @@ export interface CompositeType {
   [key: string]: any;
 }
 
+export interface PolymorphicDerivedType {
+  /**
+   * A CompositeTypeSpec that "inherits" from this CompositeTypeSpec.
+   */
```

Comment:
Haven't taken a look at other changes yet. However, I hope there is a DerivedType spec that is not polymorphic but is a simple allOf relationship, for ex: StorageAccount is an allOf Resource.

Reply:
To my knowledge there is no difference between a `PolymorphicDerivedType` and a `DerivedType`. Since we're talking about TypeSpecs, then a `PolymorphicDerivedType` is just a link from a base-type to a derived-type. All of our polymorphism relationships are allOf relationships.
Can you think of a scenario where this isn't valid?